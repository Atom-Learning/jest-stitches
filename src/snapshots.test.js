import React from 'react'
import ReactDOM from 'react-dom'
import prettyFormat from 'pretty-format'
import {createComponent, renderJson} from 'test-utils'
import serializer, {createSerializer} from 'index'

expect.addSnapshotSerializer(serializer)
afterEach(() => {
  document.getElementsByTagName('html')[0].innerHTML = ''
})

let stitchesPlugin = createSerializer()

const {ReactElement, ReactTestComponent, DOMElement} = prettyFormat.plugins

describe('jest-stitches with dom elements', () => {
  it('inserts styles into React test component snapshots', () => {
    const tree = renderJson(createComponent())

    const output = prettyFormat(tree, {
      plugins: [stitchesPlugin, ReactElement, ReactTestComponent, DOMElement],
    })

    expect(output).toMatchSnapshot()
  })

  it('inserts styles into DOM element snapshots', () => {
    const divRef = React.createRef()

    ReactDOM.render(
      React.cloneElement(createComponent(), {ref: divRef}),
      document.createElement('div')
    )

    const output = prettyFormat(divRef.current, {
      plugins: [stitchesPlugin, ReactElement, ReactTestComponent, DOMElement],
    })

    expect(output).toMatchSnapshot()
  })
})

describe('@stitches/react with DOM elements disabled', () => {
  const stitchesPlugin = createSerializer({DOMElements: false})

  it('inserts styles into React test component snapshots', () => {
    const tree = renderJson(createComponent())

    const output = prettyFormat(tree, {
      plugins: [stitchesPlugin, ReactElement, ReactTestComponent, DOMElement],
    })

    expect(output).toMatchSnapshot()
  })

  it('does not insert styles into DOM element snapshots', () => {
    const divRef = React.createRef()

    ReactDOM.render(
      React.cloneElement(createComponent(), {ref: divRef}),
      document.createElement('div')
    )

    const output = prettyFormat(divRef.current, {
      plugins: [stitchesPlugin, ReactElement, ReactTestComponent, DOMElement],
    })

    expect(output).toMatchSnapshot()
  })
})
