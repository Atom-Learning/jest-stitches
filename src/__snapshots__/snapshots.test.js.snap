// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@stitches/react with DOM elements disabled does not insert styles into DOM element snapshots 1`] = `
"<button
  class=\\"_YfjLh scid-fNnBJi\\"
>
  <svg
    class=\\"_STxcQ scid-daJnyW\\"
  />
</button>"
`;

exports[`@stitches/react with DOM elements disabled inserts styles into React test component snapshots 1`] = `
"._YfjLh {
  background-color: blue;
}

._STxcQ {
  color: yellow;
}

<button
  className={
    Object {
      \\"_className\\": \\"_YfjLh scid-fNnBJi\\",
      \\"atoms\\": Array [
        Object {
          \\"_className\\": \\"_YfjLh\\",
          \\"_isGlobal\\": undefined,
          \\"breakpoint\\": undefined,
          \\"cssHyphenProp\\": undefined,
          \\"id\\": \\"backgroundcolorinitial\\",
          \\"inlineMediaQueries\\": undefined,
          \\"pseudo\\": undefined,
          \\"selector\\": \\"\\",
          \\"toString\\": [Function toStringCachedAtom],
          \\"value\\": undefined,
          Symbol(ATOM): true,
        },
        \\"scid-fNnBJi\\",
      ],
      \\"toString\\": [Function toStringCachedAtom],
      Symbol(ATOM): true,
    }
  }
>
  <svg
    className={
      Object {
        \\"_className\\": \\"_STxcQ scid-daJnyW\\",
        \\"atoms\\": Array [
          Object {
            \\"_className\\": \\"_STxcQ\\",
            \\"_isGlobal\\": undefined,
            \\"breakpoint\\": undefined,
            \\"cssHyphenProp\\": undefined,
            \\"id\\": \\"colorinitial\\",
            \\"inlineMediaQueries\\": undefined,
            \\"pseudo\\": undefined,
            \\"selector\\": \\"\\",
            \\"toString\\": [Function toStringCachedAtom],
            \\"value\\": undefined,
            Symbol(ATOM): true,
          },
          \\"scid-daJnyW\\",
        ],
        \\"toString\\": [Function toStringCachedAtom],
        Symbol(ATOM): true,
      }
    }
  />
</button>"
`;

exports[`jest-stitches with dom elements inserts styles into DOM element snapshots 1`] = `
"._YfjLh {
  background-color: blue;
}

._STxcQ {
  color: yellow;
}

<button
  class=\\"_YfjLh scid-fNnBJi\\"
>
  <svg
    class=\\"_STxcQ scid-daJnyW\\"
  />
</button>"
`;

exports[`jest-stitches with dom elements inserts styles into React test component snapshots 1`] = `
"._YfjLh {
  background-color: blue;
}

._STxcQ {
  color: yellow;
}

<button
  className={
    Object {
      \\"_className\\": \\"_YfjLh scid-fNnBJi\\",
      \\"atoms\\": Array [
        Object {
          \\"_className\\": \\"_YfjLh\\",
          \\"_isGlobal\\": undefined,
          \\"breakpoint\\": undefined,
          \\"cssHyphenProp\\": undefined,
          \\"id\\": \\"backgroundcolorinitial\\",
          \\"inlineMediaQueries\\": undefined,
          \\"pseudo\\": undefined,
          \\"selector\\": \\"\\",
          \\"toString\\": [Function toStringCachedAtom],
          \\"value\\": undefined,
          Symbol(ATOM): true,
        },
        \\"scid-fNnBJi\\",
      ],
      \\"toString\\": [Function toStringCachedAtom],
      Symbol(ATOM): true,
    }
  }
>
  <svg
    className={
      Object {
        \\"_className\\": \\"_STxcQ scid-daJnyW\\",
        \\"atoms\\": Array [
          Object {
            \\"_className\\": \\"_STxcQ\\",
            \\"_isGlobal\\": undefined,
            \\"breakpoint\\": undefined,
            \\"cssHyphenProp\\": undefined,
            \\"id\\": \\"colorinitial\\",
            \\"inlineMediaQueries\\": undefined,
            \\"pseudo\\": undefined,
            \\"selector\\": \\"\\",
            \\"toString\\": [Function toStringCachedAtom],
            \\"value\\": undefined,
            Symbol(ATOM): true,
          },
          \\"scid-daJnyW\\",
        ],
        \\"toString\\": [Function toStringCachedAtom],
        Symbol(ATOM): true,
      }
    }
  />
</button>"
`;
